tools/create_operators_pull_request.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import logging
+import os.path
+import sys
+import tempfile
+
+from utils import (
+    run,
+    clone_repository,
+    random_short_string,
+    configure_git_user,
+    create_pull_request,
+)
+
+PROGRAM_NAME = os.path.basename(__file__)
+
+log = logging.getLogger(__name__)
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%SZ",
+)
+
+
+KUDOBUILDER_OPERATORS_REPOSITORY = "kudobuilder/operators"
+
+
+def main() -> int:
```

Comment:
This function is over 200 lines long, and the low level of abstraction of its constituents makes it hard to understand what it is doing without scrolling through and reading all of it.
Please split it up into pieces such as `parse_args`, `prepare_source_repo`, `prepare_destination_repo`, etc.

Reply:
https://github.com/mesosphere/kudo-cassandra-operator/pull/26/commits/2e3b0336c397d322b26770c090e150e7ec8d5944